### PR TITLE
Add Msg.t functions to the cooperative modules

### PR DESCRIPTION
**Pull request by hcarty**
And add missing ?more to the send function.

### CODE REVIEW

**Comment by hcarty** [comment]
`?more` was in lwt-zmq after a user submitted a PR.  It seems worth including here as well.  Sorry for missing that in my initial review!

File: zmq-deferred/src/socket.ml [hunk 1]
```diff
@@ -132,7 +132,12 @@ module Make(T: Deferred.T) = struct
   let to_socket t = t.socket
 
   let recv s = post s Receive (fun s -> ZMQ.Socket.recv ~block:false s)
-  let send s m = post s Send (fun s -> ZMQ.Socket.send ~block:false s m)
+  let send ?more s m =
+    post s Send (fun s -> ZMQ.Socket.send ?more ~block:false s m)
```
Comment (andersfugmann) [inline zmq-deferred/src/socket.ml]: I think this is dangerous. If there are multiple writers, then we will have strange interleaving between fragments and messages. 
I would advice that users only use the send_all / send_msg_all functions. We could also add call that takes a message producer something like
`val send_more: t -> (unit -> string option Deferred.t) -> unit` that would read message fragments from the given function until the function returned None. Same pattern could be used for a recv_more function.

File: zmq-deferred/src/socket.ml [hunk 2]
```diff
@@ -132,7 +132,12 @@ module Make(T: Deferred.T) = struct
...
+
+  let recv_msg s = post s Receive (fun s -> ZMQ.Socket.recv_msg ~block:false s)
+  let send_msg ?more s m =
```
Comment (andersfugmann) [inline zmq-deferred/src/socket.ml]: Same as with send_msg.

File: zmq-deferred/src/socket.ml (hunk 1 above)
Comment (hcarty) [reply zmq-deferred/src/socket.ml]: Fair point.  I think there's value in this as it was something a user created a PR for in the past.  I can pull this change out for now though and we can figure out the right way to do this in a separate PR.

Comment (hcarty) [reply zmq-deferred/src/socket.ml]: Removed in 93bf573

Comment (andersfugmann) [reply zmq-deferred/src/socket.ml]: I can be persuaded, but I don't like it. 
At least there should be a comment discouraging users from using it.

Comment (andersfugmann) [reply zmq-deferred/src/socket.ml]: You choose. I don't like breaking other peoples code either.

Comment (hcarty) [reply zmq-deferred/src/socket.ml]: I like putting it in a separate PR/discussion.  I agree that it should be discouraged in the general case.

**Pull request merged**